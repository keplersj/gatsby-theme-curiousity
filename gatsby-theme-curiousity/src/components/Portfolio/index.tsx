import React from "react";
import styled from "@emotion/styled";
import BaseLayout from "../Base";
import { PortfolioListItem as Project } from "../PortfolioListItem";
import { graphql } from "gatsby";

const Projects = styled.div`
  margin-left: 2em;
  margin-right: 2em;
`;

const ListContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
`;

interface Props {
  data: {
    allPortfolioItem: {
      edges: {
        node: {
          id: string;
          excerpt: string;
          slug: string;
          title: string;
        };
      }[];
    };
  };
}

const ProjectsPage = ({ data }: Props): React.ReactElement<Props> => (
  <BaseLayout title="Projects">
    <Projects>
      <h1>Projects</h1>
      <ListContainer>
        {data.allPortfolioItem.edges.map(
          ({ node }): React.ReactElement => (
            <Project
              key={node.id}
              location={node.slug}
              title={node.title}
              description={node.excerpt}
            />
          )
        )}
      </ListContainer>
    </Projects>
  </BaseLayout>
);

export default ProjectsPage;

export const fragment = graphql`
  fragment CuriousityPortfolio on PortfolioItem {
    id
    excerpt
    slug
    title
  }
`;
