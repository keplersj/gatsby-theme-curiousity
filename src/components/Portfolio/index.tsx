import React from "react";
import { Helmet } from "react-helmet";
import styled from "@emotion/styled";
import BaseLayout from "../Base";
import { PortfolioListItem as Project } from "../PortfolioListItem";

const Projects = styled.div`
  margin-left: 2em;
  margin-right: 2em;
`;

const ListContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
`;

interface Props {
  data: {
    allPortfolioItem: {
      edges: {
        node: {
          id: string;
          excerpt: string;
          slug: string;
          title: string;
        };
      }[];
    };
  };
}

const ProjectsPage = ({ data }: Props): React.ReactElement<Props> => (
  <>
    <Helmet>
      <script type="application/ld+json">
        {JSON.stringify({
          "@context": "http://www.schema.org",
          "@type": "CollectionPage",
          name: "Projects | Kepler Sticka-Jones",
          url: "/projects",
          breadcrumb: {
            "@type": "BreadcrumbList",
            itemListElement: [
              {
                "@type": "ListItem",
                position: 1,
                item: {
                  "@id": "https://keplersj.com/",
                  name: "Kepler Sticka-Jones"
                }
              },
              {
                "@type": "ListItem",
                position: 2,
                item: {
                  "@id": "https://keplersj.com/projects/",
                  name: "Projects"
                }
              }
            ]
          },
          about: {
            "@type": "ItemList",
            name: "Projects | Kepler Sticka-Jones",
            url: "/projects",
            numberOfItems: data.allPortfolioItem.edges.length,
            itemListElement: data.allPortfolioItem.edges.map(
              ({ node }): object => ({
                "@type": "Thing",
                name: node.title,
                description: node.excerpt,
                url: node.slug
              })
            )
          }
        })}
      </script>
    </Helmet>

    <BaseLayout title="Projects">
      <Projects>
        <h1>Projects</h1>
        <ListContainer>
          {data.allPortfolioItem.edges.map(
            ({ node }): React.ReactElement => (
              <Project
                key={node.id}
                location={node.slug}
                title={node.title}
                description={node.excerpt}
              />
            )
          )}
        </ListContainer>
      </Projects>
    </BaseLayout>
  </>
);

export default ProjectsPage;
