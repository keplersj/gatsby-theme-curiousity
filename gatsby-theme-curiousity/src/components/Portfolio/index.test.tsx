import * as React from "react";
import renderer from "react-test-renderer";
import { HelmetProvider } from "react-helmet-async";
import { useStaticQuery } from "gatsby";
import Portfolio from ".";
import deepMerge from "deepmerge";
import {
  CuriousityBaseLayoutData,
  CuriousityPortfolioPageQuery,
} from "../../__mockData__";

describe("Portfolio Page Component", () => {
  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockImplementation((): object =>
      deepMerge.all([CuriousityBaseLayoutData])
    );
  });

  it("renders as expected", () => {
    const helmetContext: { helmet?: object } = {};
    const tree = renderer
      .create(
        <HelmetProvider context={helmetContext}>
          <Portfolio data={CuriousityPortfolioPageQuery} />
        </HelmetProvider>
      )
      .toJSON();

    expect(helmetContext.helmet).toMatchSnapshot();
    expect(tree).toMatchSnapshot();
  });
});
