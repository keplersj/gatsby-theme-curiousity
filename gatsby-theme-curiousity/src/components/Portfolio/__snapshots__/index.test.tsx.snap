// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Page Component renders as expected 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      content="Projects | Test Site"
      data-rh={true}
      property="og:title"
    />
    <meta
      content="https://example.dev/"
      data-rh={true}
      property="og:url"
    />
    <title
      data-rh={true}
    >
      Projects | Test Site
    </title>
  </head>
  <body />
</html>
`;

exports[`Portfolio Page Component renders as expected 2`] = `
.emotion-4 {
  margin-left: 2em;
  margin-right: 2em;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.emotion-2 {
  margin: 1em;
  background-color: rgba(var(--starstuff-card-default-background-color),0.5);
  border-radius: 10px;
  box-shadow: 0.5em 0.5em 0.75em rgba(var(--starstuff-card-default-shadow-color),0.3);
  -webkit-transition: box-shadow 1s;
  transition: box-shadow 1s;
  --starstuff-card-default-background-color: 245,245,245;
  --starstuff-card-default-shadow-color: 10,10,10;
}

.emotion-2:active,
.emotion-2:focus,
.emotion-2:focus-within,
.emotion-2:hover {
  box-shadow: 0.8em 0.8em 1em rgba(var(--starstuff-card-default-shadow-color),0.4);
}

@media screen and (prefers-color-scheme:dark) {
  .emotion-2 {
    --starstuff-card-default-background-color: 55,55,55;
  }
}

@media screen and (min-width:1024px) {
  .emotion-2 {
    min-height: 192px;
  }
}

@media screen and (max-width:512px) {
  .emotion-2 {
    max-width: 100vw;
  }
}

@media screen and (min-width:768px) {
  .emotion-2 {
    max-width: 33vw;
  }
}

@media screen and (min-width:1024px) {
  .emotion-2 {
    max-width: 25vw;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<main>
  <div
    className="emotion-4"
  >
    <h1>
      Projects
    </h1>
    <div
      className="emotion-3"
    >
      <div
        className="emotion-2"
      >
        <article
          className="emotion-1"
        >
          <script
            type="application/ld+json"
          >
            Object {
            "@context": "https://schema.org",
            "@id": "https://example.dev/",
            "@type": "CreativeWork",
            "headline": "",
            "mainEntityOfPage": "https://example.dev/",
            "name": "",
            "url": "https://example.dev/",
          }
          </script>
          <div
            className="emotion-0"
          >
            <a
              href=""
            >
              <h2>
                
              </h2>
            </a>
            <p>
              
            </p>
          </div>
        </article>
      </div>
    </div>
  </div>
</main>
`;
